crates/ruff_server/src/session/settings.rs

```diff
@@ -1,25 +1,452 @@
+                workspace_settings
+                    .get(&url)
+                    .expect("workspace setting should exist"),
+                &global_settings
+            ),
+            ResolvedClientSettings {
+                fix_all: true,
+                organize_imports: true,
+                lint_enable: true,
+                disable_rule_comment_enable: false,
+                fix_violation_enable: false,
+            }
+        );
+        let url = Url::parse("file:///Users/test/projects/scipy").expect("url should parse");
+        assert_eq!(
+            ResolvedClientSettings::with_workspace(
+                workspace_settings
+                    .get(&url)
+                    .expect("workspace setting should exist"),
+                &global_settings
+            ),
+            ResolvedClientSettings {
+                fix_all: true,
+                organize_imports: true,
+                lint_enable: true,
+                disable_rule_comment_enable: true,
+                fix_violation_enable: false,
+            }
+        );
+    }
+
+    #[test]
+    fn test_global_only_init_options_deserialize() {
+        let options: InitializationOptions = fixture!("settings/global_only.json");
+
+        assert_debug_snapshot!(options, @r###"
+        GlobalOnly {
+            settings: Some(
+                ClientSettings {
+                    fix_all: Some(
+                        false,
+                    ),
+                    organize_imports: None,
+                    lint: Some(
+                        Lint {
+                            enable: None,
+                        },
+                    ),
+                    code_action: Some(
+                        CodeAction {
+                            disable_rule_comment: Some(
+                                CodeActionSettings {
+                                    enable: Some(
+                                        false,
+                                    ),
+                                },
+                            ),
+                            fix_violation: None,
+                        },
+                    ),
+                },
+            ),
         }
+        "###);
+    }
+
+    #[test]
+    fn test_global_only_resolves_correctly() {
+        let options = fixture!("settings/global_only.json");
+
+        let AllSettings {
+            global_settings, ..
+        } = AllSettings::from_init_options(options);
+        assert_eq!(
+            ResolvedClientSettings::global_only(&global_settings),
+            ResolvedClientSettings {
+                fix_all: false,
+                organize_imports: true,
+                lint_enable: true,
+                disable_rule_comment_enable: false,
+                fix_violation_enable: true,
+            }
+        );
+    }
+
+    #[test]
+    fn test_empty_init_options_deserialize() {
+        let _: InitializationOptions = fixture!("settings/empty.json");
```

Comment:
Is it intentional that we don't assert to what the empty options deserialize to? Like we could add a `assert_eq(deserialized, InitialzationOptions::default())` to at least compare to something.

Reply:
No, that was an oversight. Let's add that.